src/bindings/html/view.js

```diff
@@ -24,23 +24,27 @@ export class View {
       'ar-x-psbidi': createPseudo(this, 'ar-x-psbidi')
     };
 
-    this._interactive = documentReady().then(
-      () => init(this, client));
+    const initialized = documentReady().then(() => init(this, client));
+    this._interactive = initialized.then(() => client);
+    this.ready = initialized.then(langs => translateDocument(this, langs));
 
     const observer = new MutationObserver(onMutations.bind(this));
     this._observe = () => observer.observe(doc, observerConfig);
     this._disconnect = () => observer.disconnect();
 
-    const translateView = langs => translateDocument(this, langs);
-    client.on('translateDocument', translateView);
-    this.ready = this._interactive.then(
-      client => client.method('resolvedLanguages')).then(
-      translateView);
```

Comment:
Removing this double communication with the service (first `this._interactive` calls `registerView`, then this code calls `resolvedLanguages`) is what resulted in the performance win.